### ISSUE TITLE
Detection 目标检测

### ISSUE THREAD

**Issue by Bryce1010** [Bryce1010/DeepLearning-Project#11]
## 综述  
-  STOA papers with code [[url]](https://paperswithcode.com/sota/object-detection-on-coco)  
- 【DeepLearning综述】 DeepLearning Detection 综述 [[blog]](https://bryce1010.blog.csdn.net/article/details/103892627)
-   MMDetection: Open MMLab Detection Toolbox and Benchmark [[paper]](https://arxiv.org/pdf/1906.07155.pdf)  
    -  香港中文大学，商汤等联合提出的MMDetection，包括检测模型，实体分割等state-of-art模型框架源码，属业界良心。
- [2019.07] A Survey of Deep Learning-based Object Detection [[paper]](https://arxiv.org/pdf/1907.09408.pdf)  
    - 西安电子科技大学关于目标检测的论文综述。  
- [2019.08]Recent Advances in Deep Learning for Object Detection [[paper]](https://arxiv.org/pdf/1908.03673.pdf)  
    - 新加坡管理大学论文综述。
- [2019.09]Imbalance Problems in Object Detection: A Review  [[paper]](https://arxiv.org/pdf/1909.00169.pdf) [[github]](https://github.com/kemaloksuz/ObjectDetectionImbalance)   
    - 中东科技大学(Middle East Technical University)一篇关于目标检测领域imbalance problem的综述。imbalance problem包括Class imbalance， Scale imbalance，Spatial imbalance， objective imbalance。论文对各个方面进行归纳，提出问题和分析解决方案。
- (2019.05) Object Detection in 20 Years: A Survey  [[paper]](https://arxiv.org/pdf/1905.05055.pdf)    

- (2020.02) DEEP DOMAIN ADAPTIVE OBJECT DETECTION: A SURVEY [[paper]](https://arxiv.org/pdf/2002.06797.pdf)  

- deep learning object detection [[github]](https://github.com/hoya012/deep_learning_object_detection)  
    > A paper list of object detection using deep learning.
## 书籍  
- [ ] 视觉计算基础：计算机视觉、图形学和图像处理的核心概念 
- [ ] 计算机视觉――一种现代方法（第二版）
- [ ] 数字图像目标检测与识别―理论与实践 [Object Detection and Recognition in Digital Images]    
- [ ] 目标检测好文分享 [[zihu]](https://zhuanlan.zhihu.com/p/140036646)

**Comment by Bryce1010** [comment]
## One-Stage  
- [ ] (2019.04) FCOS: Fully Convolutional One-Stage Object Detection [[paper]](https://arxiv.org/pdf/1904.01355.pdf)    
- [ ] (2019.04) HAR-Net: Joint Learning of Hybrid Attention for Single-stage Object Detection  [[paper]](https://arxiv.org/pdf/1904.11141.pdf)
- [ ] (2019.04) CenterNet: Object Detection with Keypoint Triplets [[paper]](https://arxiv.org/pdf/1904.08189.pdf)

**Comment by Bryce1010** [comment]
## Tricks

**Comment by Bryce1010** [comment]
## todo list

**Comment by Bryce1010** [comment]
## 目标检测比赛中的tricks  
https://zhuanlan.zhihu.com/p/102817180

**Comment by Bryce1010** [comment]
## tricks in MMdetection   

- [ ] Mixed precision training 
- [ ] soft NMS 
- [ ] OHEM  
- [ ] DCN(Deformable convolution network)  
- [ ] DCN v2
- [ ] Train from scratch  
- [ ] ScratchDet
- [ ] M2Det
- [ ] GCNet  
- [ ] Genelized Attention  
- [ ] Sync BN  
- [ ] Group Normalization   
- [ ] HRNet  
- [ ] Guided Anchoring  
- [ ] Libra R-CNN (imbalanced problem)

**Comment by Bryce1010** [comment]
### NMS算法过程  
（1）将所有框的得分排序，选中最高分及其对应的框
（2）遍历其余的框，如果和当前最高分框的重叠面积(IOU)大于一定阈值，我们就将框删除。
（3）从未处理的框中继续选一个得分最高的，重复上述过程。
![image](https://user-images.githubusercontent.com/30361513/82114894-fb46e180-9791-11ea-81bc-276d911ee113.png)

```python
# python3
import numpy as np

def py_nms(dets, thresh):
    """Pure Python NMS baseline."""
    #x1、y1、x2、y2、以及score赋值
    x1 = dets[:, 0]
    y1 = dets[:, 1]
    x2 = dets[:, 2]
    y2 = dets[:, 3]
    scores = dets[:, 4]

    #每一个候选框的面积
    areas = (x2 - x1 + 1) * (y2 - y1 + 1)
    #order是按照score降序排序的
    order = scores.argsort()[::-1]

    keep = []
    while order.size > 0:
        i = order[0]
        keep.append(i)
        #计算当前概率最大矩形框与其他矩形框的相交框的坐标，会用到numpy的broadcast机制，得到的是向量
        xx1 = np.maximum(x1[i], x1[order[1:]])
        yy1 = np.maximum(y1[i], y1[order[1:]])
        xx2 = np.minimum(x2[i], x2[order[1:]])
        yy2 = np.minimum(y2[i], y2[order[1:]])

        #计算相交框的面积,注意矩形框不相交时w或h算出来会是负数，用0代替
        w = np.maximum(0.0, xx2 - xx1 + 1)
        h = np.maximum(0.0, yy2 - yy1 + 1)
        inter = w * h
        #计算重叠度IOU：重叠面积/（面积1+面积2-重叠面积）
        ovr = inter / (areas[i] + areas[order[1:]] - inter)

        #找到重叠度不高于阈值的矩形框索引
        inds = np.where(ovr <= thresh)[0]
        #将order序列更新，由于前面得到的矩形框索引要比矩形框在原order序列中的索引小1，所以要把这个1加回来
        order = order[inds + 1]
    return keep

# test
if __name__ == "__main__":
    dets = np.array([[30, 20, 230, 200, 1], 
                     [50, 50, 260, 220, 0.9],
                     [210, 30, 420, 5, 0.8],
                     [430, 280, 460, 360, 0.7]])
    thresh = 0.35
    keep_dets = py_nms(dets, thresh)
    print(keep_dets)
    print(dets[keep_dets])


```

**Comment by Bryce1010** [comment]
## 5.20 detection 调研  

survy
- Imbalance Problems in Object Detection: A Review



one-stage 
- YOLOv4: Optimal Speed and Accuracy of Object Detection
- MatrixNets: A New Scale and Aspect Ratio Aware Architecture for Object Detection


two stage 
+ Gliding vertex on the horizontal bounding box for multi-oriented object detection
- 1st Place Solutions for OpenImage2019 -- Object Detection and Instance Segmentation
- ResNeSt: Split-Attention Networks

- Dynamic R-CNN: Towards High Quality Object Detection via Dynamic Training
- Revisiting the Sibling Head in Object Detector

anchor free
- CentripetalNet: Pursuing High-quality Keypoint Pairs for Object Detection
- (2019NIPS) FreeAnchor: Learning to Match Anchors for Visual Object Detection

nas
- Fast Neural Network Adaptation via Parameter Remapping and Architecture Search



neck 
- (2019NIPS)Cascade RPN: Delving into High-Quality Region Proposal Network with Adaptive Convolution

unknown
- Fooling Detection Alone is Not Enough: Adversarial Attack against Multiple Object Tracking  
+ Bridging the Gap Between Anchor-based and Anchor-free Detection via Adaptive Training Sample Selection
+ (2019NIPS) Memory-oriented Decoder for Light Field Salient Object Detection  


loss function  
Distance-IoU Loss: Faster and Better Learning for Bounding Box Regression  


activation function  
- Mish: A Self Regularized Non-Monotonic Neural Activation Function


semi-supervised 
- A Simple Semi-Supervised Learning Framework for Object Detection  [[paper]](https://paperswithcode.com/paper/a-simple-semi-supervised-learning-framework)  
+ (2019 NIPS) Consistency-based Semi-supervised Learning for Object detection  




weakly supervised
- Instance-aware, Context-focused, and Memory-efficient Weakly Supervised Object Detection  
- Object Instance Mining for Weakly Supervised Object Detection

few-shot  
- Frustratingly Simple Few-Shot Object Detection
- (2019NIPS) One-Shot Object Detection with Co-Attention and Co-Excitation  
- (2019ICCV) Meta R-CNN: Towards General Solver for Instance-Level Low-Shot Learning  



3D object detection  
- Disp R-CNN: Stereo 3D Object Detection via Shape Prior Guided Instance Disparity Estimation
- Point-GNN: Graph Neural Network for 3D Object Detection in a Point Cloud
- Pseudo-LiDAR++: Accurate Depth for 3D Object Detection in Autonomous Driving
- PV-RCNN: Point-Voxel Feature Set Abstraction for 3D Object Detection  
- Deep Learning for 3D Point Clouds: A Survey  






tracking  
- A Simple Baseline for Multi-Object Tracking
- Tracking Objects as Points  




segmentation  
- SOLOv2: Dynamic, Faster and Stronger
- RDSNet: A New Deep Architecture for Reciprocal Object Detection and Instance Segmentation
- CenterMask : Real-Time Anchor-Free Instance Segmentation



augmentation 
- GridMask Data Augmentation
- RandAugment: Practical automated data augmentation with a reduced search space

backbone 
- SpineNet: Learning Scale-Permuted Backbone for Recognition and Localization
+ Stand-Alone Self-Attention in Vision Models
+ (2019 NIPS) Unsupervised Learning of Object Keypoints for Perception and Control
+ (2019 NIPS) FreeAnchor: Learning to Match Anchors for Visual Object Detection
+ (2019NIPS) DetNAS: Backbone Search for Object Detection  
+ CSPNet: A New Backbone that can Enhance Learning Capability of CNN
+ (2020 ICLR)Deformable Kernels: Adapting Effective Receptive Fields for Object Deformation
- (2019 ICCV) Enriched Feature Guided Refinement Network for Object Detection
- (2019ICCV) Scaling Object Detection by Transferring Classification Weights
+ CBNet: A Novel Composite Backbone Network Architecture for Object Detection
+ (HRNet) Deep High-Resolution Representation Learning for Visual Recognition

domain adaptation  
- (2020 ICLR) SCL: Towards Accurate Domain Adaptive Object Detection via Gradient Detach Based Stacked Complementary Losses

**Comment by Bryce1010** [comment]
## Recent Detection papers   
### 2020
- [Spiking-YOLO] Spiking-YOLO: Spiking Neural Network for Real-time Object Detection | [AAAI' 20]  
    > 利用脉冲神经网络做目标检测；并提出一个channel-wise Normalization 和 signed neuron with imbalanced threshold。   
    > 没看懂spiking neural  

- Tell Me What They're Holding: Weakly-supervised Object Detection with Transferable Knowledge from Human-object Interaction | [AAAI' 20]
